PostHog/Classes/PHGPayloadManager.m

```diff
@@ -102,15 +116,11 @@ - (void)identify:(NSString *)distinctId properties:(NSDictionary *)properties op
 {
     NSCAssert1(distinctId.length > 0, @"distinctId (%@) must not be empty.", distinctId);
 
-    NSString *anonymousId = [options objectForKey:@"$anon_distinct_id"];
-    if (anonymousId) {
-        [self saveAnonymousId:anonymousId];
-    } else {
-        anonymousId = self.cachedAnonymousId;
-    }
+    [self saveAnonymousId:self.integration.distinctId];
+    [self.integration saveDistinctId:distinctId];
```

Comment:
Asking here as a lot of this logic is new / confusing to me 😅 

Why do we save the anonymousId to be the previous distinctId and then the distinctId to be the new one? 

More specifically I'm wondering why we need two IDs? Why is the flow not like
* Load or generate distinctID
* use ID everywhere for `distinct_id` 
* on Identify payload set the current `distinct_id` as `$anon_distinct_id` and the new one as `distinct_id`
  * save the new `distinct_id` to the persisted ID.

I'm sure there is a reason we have two separate IDs but I'm just not sure what it is 😅

Reply:
The primary reason I adhered to shifting old distinct_id to anon_distinct_id when id'ing is to satisfy feature flag continuity if necessary. We recently adjusted the logic such that anon_distinct_id is also passed during feature flag calls to track when experience continuity needs to happen. Apart from that, I would like to try to keep the client libraries in alignment. This seemed like a simple approach to do that but definitely isn't following posthog-js perfectly